dask/dataframe/core.py

```diff
@@ -540,9 +540,79 @@ def sample(self, frac, replace=False, random_state=None):
         return self._constructor(merge(self.dask, dsk), name,
                                  self._pd, self.divisions)
 
-    @derived_from(pd.DataFrame)
     def to_hdf(self, path_or_buf, key, mode='a', append=False, complevel=0,
                complib=None, fletcher32=False, get=None, **kwargs):
+        """ Export frame to hdf file(s)
+
+        Export dataframe to one or multiple hdf5 files or nodes.
+
+        Exporting dataframe to multiple targets enables improved parallelism
+        and is expected to provide better performance for longer computations.
+
+        Parameters
+        ----------
+        path_or_buf: HDFStore object or string
+            Destination file(s). If string, can contain a single wildcard
+            to save each partition to a different file. Only one widlcard
+            is allowed in both path and key.
+        key: string
+            A node / group identifier, can contain a single wildcard
+            to save each partition to a different hdf node in a single
+            file. This enables better parallelism compared to saving
+            entire dataframe in a single node.
```

Comment:
Perhaps include an example showcasing the use of path_or_buf + key with wildcards? From the description is a bit confusing if only one wildcard is allowed in total, or you can have one wildcard in path_or_buf and another one in key.

I assume that it's either one wildcard for path_or_buf or one wilcard for key, since you want to save partitions in either different hdf files or same hdf file but different nodes. Right? We can say this in the second paragraph of the method docstring above.